inline.h

```diff
@@ -3090,6 +3090,94 @@ Perl_utf8_to_uvchr_buf_helper(pTHX_ const U8 *s, const U8 *send, STRLEN *retlen)
     }
 }
 
+/*
+=for apidoc utf8_to_uv
+
+Convert the first character in the UTF-8-encoded string starting at C<s> to the
+native code point it represents, storing its value into C<*cp>, and the number
+of bytes of C<s> it occupies into C<*advance>.
+
+Only the bytes from C<s> to S<C<send -1>> are examined.  B<It is an error to call
+this function without> S<C<send E<gt> s>>.
+
+If the bytes starting at C<s> do not represent at least one well-formed UTF-8
+character, this function returns C<false> and sets C<*cp> to the Unicode
+C<REPLACEMENT CHARACTER>.
+
+Otherwise it returns C<true>.
+
+With both success and failure, the value of C<*advance> is the number of bytes
+to add to C<s> to look for the next code point in it.  You might be tempted to
+use C<UTF8_SKIP> instead for this purpose, but B<don't>; it requires extra
+work, and is likely wrong when this function returns C<false>.
+
+Starting in Perl v5.42, this is the preferred function to call to convert from
+UTF-8 for code that doesn't want to get into the weeds of possible
+malformations.
+
+=cut
+*/
+
+PERL_STATIC_INLINE  bool
+Perl_utf8_to_uv(pTHX_ const U8 *s, const U8 *send, UV *cp, STRLEN *advance)
+{
+    PERL_ARGS_ASSERT_UTF8_TO_UV;
+
+    return utf8_to_uv_flags(s, send, cp, advance, 0) >= 0;
+}
+
+/*
+=for apidoc utf8_to_uv_flags
+
+Convert the first character in the UTF-8-encoded string starting at C<s> to the
+native code point it represents, storing its value into C<*cp>, and the number
+of bytes of C<s> it occupies into C<*advance>.
+
+Only the bytes from C<s> to S<C<send -1>> are examined.  B<It is an error to call
+this function without> S<C<send E<gt> s>>.
+
+If the bytes starting at C<s> do not represent at least one well-formed UTF-8
+character, this function returns a negative value and sets C<*cp> to the
+Unicode C<REPLACEMENT CHARACTER>.
+
+Otherwise it returns a value greater than or equal to zero.
+
+With both success and failure, the value of C<*advance> is the number of bytes
+to add to C<s> to look for the next code point in it.  You might be tempted to
+use C<UTF8_SKIP> instead for this purpose, but B<don't>; it requires extra
+work, and is likely wrong when this function returns a negative value.
+
+You can pass in the C<flags> parameter any of the C<DISALLOW>-type flags
+accepted by C<L</utf8n_to_uvchr>> to restrict the types of code points accepted
+by this function.
+
+A negative return from this function indicates an error.  If you take its
+negative (turning it into a positive), the value will be the same as that
+returned in C<*error> parameter to C<L</utf8n_to_uvchr_error>>, so you can
+determine the exact malformations.
+
+This function is an alternative to that function, but with an API more
+consistent with that of C<L</utf8_to_uv>>.
+
```

Comment:
Does this warn on malformed input when utf8 warnings are enabled in scope?

If it does, can I turn that off?

Reply:
Most code wants to not have to concern itself with malformations.  But all code needs to be protected from them.  This is designed for the vast majority case where malformations are not expected, and don't want to be checked for.

For code that does need more control, there is the family of `utf8n_to_uvchr` functions, documented in `perlapi`   which gives complete flexibility to do whatever you want.

Reply:
I realized, I hadn't answered your question.  Warnings are whatever are in effect in the lexical scope.  A parameter could be added or a variant function could be written to turn them off regardless of their being on.